### ISSUE TITLE
music.play(pin=None) causes typeError, expecting a pin

### ISSUE THREAD

**Issue by martinwork** [bbcmicrobit/micropython#713]
Testing with https://python.microbit.org/v/beta
Editor Version: 2.2.0-beta.5;
MicroPython Version: 1.0.1/2.0.0-beta.5

An error occurs if music.play specifies pin=None, either directly or through a variable.

```python
import microbit
from microbit import *
import music

myPin=pin0

while True:
    music.play(music.JUMP_UP)
    sleep(1000)
    
    music.play(music.JUMP_UP,pin=myPin)
    sleep(1000)

    myPin=None
```

**Comment by microbit-carlos** [comment]
Good catch, thanks Martin!
For compatibility with V2 we should to accept `None` as a `pin` value in V1 as well.

**Comment by dpgeorge** [comment]
Fixed by 1252f887ddc790676bf9314a136bd17650b9c36c
